sources/Interop/Windows/shared/uuids/Windows.Manual.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+// Ported from shared/uuids.h in the Windows SDK for Windows 10.0.19041.0
+// Original source is Copyright © Microsoft. All rights reserved.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.Contracts;
+using System.Runtime.CompilerServices;
+
+namespace TerraFX.Interop
+{
+    public static partial class Windows
+    {
+        /// <summary>
+        /// Retrieves the GUID of of a specified type.
+        /// </summary>
+        /// <typeparam name="T">The type to retrieve the GUID for.</typeparam>
+        /// <returns>
+        /// A <see cref="GUID"/> value wrapping a pointer to the GUID data for the
+        /// input type. This value can be either converted to a <see cref="Guid"/>
+        /// pointer, or implicitly assigned to a <see cref="Guid"/> value.
+        /// </returns>
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe GUID __uuidof<T>()
+            where T : unmanaged
+        {
+            return new(UUID<T>.RIID);
+        }
+
+        /// <summary>
+        /// A proxy type that wraps a pointer to GUID data. Values of this type can be implicitly
+        /// converted to and assigned to <see cref="Guid"/>* or <see cref="Guid"/> parameters.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public readonly unsafe ref struct GUID
+        {
+            private readonly Guid* riid;
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal GUID(Guid* riid)
+            {
+                this.riid = riid;
+            }
+
+            /// <summary>
+            /// Reads a <see cref="Guid"/> value from the GUID buffer for a given <see cref="GUID"/> instance.
+            /// </summary>
+            /// <param name="guid">The input <see cref="GUID"/> instance to read data for.</param>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static implicit operator Guid(GUID guid) => *guid.riid;
+
+            /// <summary>
+            /// Returns the <see cref="Guid"/>* pointer to the GUID buffer for a given <see cref="GUID"/> instance.
+            /// </summary>
+            /// <param name="guid">The input <see cref="GUID"/> instance to read data for.</param>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static implicit operator Guid*(GUID guid) => guid.riid;
```

Comment:
What does the codegen look like for this and the other implicit operator?

Reply:
I'd say it looks quite nice 😄
Plus there's the fact that the tiered JIT could even elide the static constructor check entirely.
But other than this there's basically just the call to the static constructor (which we would've had anyway since `Windows` has statically initialized fields as well), and then all the rest is completely elided by the JIT.

<details>
  <summary><b>Guid* converter (click to expand):</b></summary>
  <br/>

```csharp
public static unsafe Guid* GetAddr()
{
    return __uuidof<ID3D12SomeType>();
}
```
```asm
sub      rsp, 40
mov      rcx, 0xD1FFAB1E
xor      edx, edx
call     CORINFO_HELP_CLASSINIT_SHARED_DYNAMICCLASS
mov      rax, 0xD1FFAB1E
mov      rax, qword ptr [rax]
add      rsp, 40
ret
```
</details>

<details>
  <summary><b>Guid converter (click to expand):</b></summary>
  <br/>

```csharp
public static unsafe Guid GetGuid()
{
    return Windows.__uuidof<ID3D12Test>();
}
```
```asm
push     rsi
sub      rsp, 32
vzeroupper 
mov      rsi, rcx
mov      rcx, 0xD1FFAB1E
xor      edx, edx
call     CORINFO_HELP_CLASSINIT_SHARED_DYNAMICCLASS
mov      rax, 0xD1FFAB1E
mov      rax, qword ptr [rax]
vmovdqu  xmm0, xmmword ptr [rax]
vmovdqu  xmmword ptr [rsi], xmm0
mov      rax, rsi
add      rsp, 32
pop      rsi
ret  
```
</details>

In particular for this as we have a direct comparison, the codegen is slightly smaller than just suing static GUIDs from the `Windows` class, but of course this approach is still much easier to use and less error prone.

<details>
  <summary><b>Accessing Windows.IID_Foo (click to expand):</b></summary>
  <br/>

```csharp
public static unsafe Guid GetGuid()
{
    return Windows.IID_Foo;
}
```
```asm
push     rsi
sub      rsp, 32
vzeroupper 
mov      rsi, rcx
mov      rcx, 0xD1FFAB1E
mov      edx, 5
call     CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
mov      rax, 0xD1FFAB1E
mov      rax, gword ptr [rax]
vmovdqu  xmm0, xmmword ptr [rax+8]
vmovdqu  xmmword ptr [rsi], xmm0
mov      rax, rsi
add      rsp, 32
pop      rsi
ret   
```
</details>

Considering that here we also have the prologue/epilogue that'd just go away once this method is just inlined 🚀